cli/run/loadConfigFile.ts

```diff
@@ -173,3 +204,55 @@ async function getConfigList(configFileExport: any, commandOptions: any): Promis
 	}
 	return Array.isArray(config) ? config : [config];
 }
+
+function tryResolveIdFromLoadedPlugins(
+	warnings: BatchWarnings,
+	id: string,
+	importer: string | undefined,
+	plugins: readonly Plugin<any>[]
+) {
+	for (const { resolveId, name } of plugins) {
+		const handler = typeof resolveId === 'function' ? resolveId : resolveId?.handler;
+		if (handler) {
+			// Since we don't support promises (because externality check is not defined as supporting them), we don't support plugin context either
+			// There probably is some way to do that (maybe save each plugin context by wrapping its buildStart hook?), but for now, let's just omit that.
+			let resolveIdResult: ResolveIdResult | Promise<ResolveIdResult>;
+			try {
+				resolveIdResult = handler.bind(undefined!)(id, importer, {
+					attributes: {},
+					isEntry: !importer
+				});
+			} catch (error) {
+				warnings.add({
+					cause: error,
+					id,
+					message: 'Error when resolving id for config file transformation',
+					plugin: name
+				});
+			}
+			if (resolveIdResult) {
+				if (typeof resolveIdResult === 'string') {
+					id = resolveIdResult;
+					break;
+				} else if (typeof resolveIdResult === 'object' && !(resolveIdResult instanceof Promise)) {
+					const external = resolveIdResult.external;
+					if (typeof external === 'boolean') {
+						return external;
+					} else {
+						id = resolveIdResult.id;
+						break;
+					}
+				} else {
+					// Throw an error instead of just 'losing' a promise.
+					// But would warning be ok?
+					throw new Error(`Plugin "${name}" returned unsupported value from its "resolveId" hook.`);
```

Comment:
Is this okay way to handle potential asyncness of `resolveId` hook? Since externality check itself can not be async.

Reply:
Went the other way as suggestd by reviewer, so this is not relevant anymore 👍